## Step 1: Load and explore data
* Load the training datasets that contain the following information: Categorical, Quantitative, Connectome metrices, and training labels.
* Merge all data on common columns
* Identify and handle missing values, while standardise the formats of data

In [1]:
# Import used libraries
import pandas as pd

In [10]:
# Load all training datasets
train_categorical = pd.read_excel("widsdatathon2025/TRAIN/TRAIN_CATEGORICAL_METADATA.xlsx")
train_connectome = pd.read_csv("widsdatathon2025/TRAIN/TRAIN_FUNCTIONAL_CONNECTOME_MATRICES.csv")
train_quantitative = pd.read_excel("widsdatathon2025/TRAIN/TRAIN_QUANTITATIVE_METADATA.xlsx")
train_labels = pd.read_excel("widsdatathon2025/TRAIN/TRAINING_SOLUTIONS.xlsx")


In [12]:
# Examine categorical data
print(train_categorical.head())
print(train_categorical.info())
print(train_categorical.shape)

  participant_id  Basic_Demos_Enroll_Year  Basic_Demos_Study_Site  \
0   UmrK0vMLopoR                     2016                       1   
1   CPaeQkhcjg7d                     2019                       3   
2   Nb4EetVPm3gs                     2016                       1   
3   p4vPhVu91o4b                     2018                       3   
4   M09PXs7arQ5E                     2019                       3   

   PreInt_Demos_Fam_Child_Ethnicity  PreInt_Demos_Fam_Child_Race  \
0                               0.0                            0   
1                               1.0                            2   
2                               1.0                            8   
3                               0.0                            8   
4                               0.0                            1   

   MRI_Track_Scan_Location  Barratt_Barratt_P1_Edu  Barratt_Barratt_P1_Occ  \
0                        1                      21                      45   
1                   

In [13]:
# fMRI data
print(train_connectome.head())
print(train_connectome.info())
print(train_connectome.shape)

  participant_id  0throw_1thcolumn  0throw_2thcolumn  0throw_3thcolumn  \
0   70z8Q2xdTXM3          0.093473          0.146902          0.067893   
1   WHWymJu6zNZi          0.029580          0.179323          0.112933   
2   4PAQp1M6EyAo         -0.051580          0.139734          0.068295   
3   obEacy4Of68I          0.016273          0.204702          0.115980   
4   s7WzzDcmDOhF          0.065771          0.098714          0.097604   

   0throw_4thcolumn  0throw_5thcolumn  0throw_6thcolumn  0throw_7thcolumn  \
0          0.015141          0.070221          0.063997          0.055382   
1          0.038291          0.104899          0.064250          0.008488   
2          0.046991          0.111085          0.026978          0.151377   
3          0.043103          0.056431          0.057615          0.055773   
4          0.112988          0.071139          0.085607          0.019392   

   0throw_8thcolumn  0throw_9thcolumn  ...  195throw_196thcolumn  \
0         -0.035335     

In [14]:
# Quantitative data
print(train_quantitative.head())
print(train_quantitative.info())
print(train_quantitative.shape)

  participant_id  EHQ_EHQ_Total  ColorVision_CV_Score  APQ_P_APQ_P_CP  \
0   UmrK0vMLopoR          40.00                    13               3   
1   CPaeQkhcjg7d         -94.47                    14               3   
2   Nb4EetVPm3gs         -46.67                    14               4   
3   p4vPhVu91o4b         -26.68                    10               5   
4   M09PXs7arQ5E           0.00                    14               5   

   APQ_P_APQ_P_ID  APQ_P_APQ_P_INV  APQ_P_APQ_P_OPD  APQ_P_APQ_P_PM  \
0              10               47               13              11   
1              13               34               18              23   
2              10               35               16              10   
3              12               39               19              16   
4              15               40               20              24   

   APQ_P_APQ_P_PP  SDQ_SDQ_Conduct_Problems  SDQ_SDQ_Difficulties_Total  \
0              28                         0                

In [15]:
# Labels
print(train_labels.head())
print(train_labels.info())
print(train_labels.shape)

  participant_id  ADHD_Outcome  Sex_F
0   UmrK0vMLopoR             1      1
1   CPaeQkhcjg7d             1      0
2   Nb4EetVPm3gs             1      0
3   p4vPhVu91o4b             1      1
4   M09PXs7arQ5E             1      1
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1213 entries, 0 to 1212
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   participant_id  1213 non-null   object
 1   ADHD_Outcome    1213 non-null   int64 
 2   Sex_F           1213 non-null   int64 
dtypes: int64(2), object(1)
memory usage: 28.6+ KB
None
(1213, 3)


In [7]:
# Merge datasets on the column 'participant_id'
train_data = pd.merge(train_categorical, train_quantitative, on='participant_id')
train_data = pd.merge(train_data, train_connectome, on='participant_id')
train_data = train_data.merge(train_labels, on='participant_id')

# Delete irrelevant column
train_data.drop(['participant_id'], axis=1, inplace=True)

In [8]:
# Identify missing values
missing_values = train_data.isnull().sum().sort_values(ascending=False)
print("Missing Values:\n", missing_values[missing_values > 0])

Missing Values:
 MRI_Track_Age_at_Scan               360
PreInt_Demos_Fam_Child_Ethnicity     11
dtype: int64


#### **Summary of the Dataset:**
**Categorical Metadata (10 columns, 1213 entries)**

* Contains data regarding study site information, as well as study subject's family background, and demographic data such as ethnicity
* The PreInt_Demos_Fam_Child_Ethnicity column contains 11 missing values to be handled

**Connectime Metrices (19901 columns, 1213 entries)**

* Contains fMRI Data that is recorded

**Quantitative Metadata (19 columns, 1213 entries)**

* Contains data regarding behavioural test scores
* The MRI_track_Age_at_Scan column contains 360 missing values to be handled

**Training Solutions (3 columns, 1213 entries)**

* Contains data regarding their final diagnosis of ADHD and study subject's gender

### Step 2: Preprocess the data

### Step 3: Model selection

### Step 4: Train the model

### Step 5: Evaluate the model

### Step 6: Hyperparameter tuning